### ISSUE TITLE
Error decoding the received TLS packet

### ISSUE THREAD

**Issue by Perflyst** [omarroth/invidious#782]
Wanted to watch this video https://fi.invidious.snopyta.org/watch?v=5ZosFi-IzRU or on https://de.invidious.snopyta.org/watch?v=5ZosFi-IzRU but it did not load. Chromium reported `failed net::ERR_CONTENT_LENGHT_MISMTACH` and if I try via wget the following errors occur:

`A TLS packet with unexpected length was received.`
`Error decoding the received TLS packet.`
`The TLS connection was non-properly terminated.`

Logs below:

<details>
<br>

```
$ wget "https://r4---sn-h0jeenek.googlevideo.com/videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&fexp=23842630&c=WEB&txp=3416222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cmime%2Cratebypass%2Cdur%2Clmt&sig=ALgxI2wwRQIhANqYc5fuRTlzb0Lzrivm5_XMLC9arLPxDFM3daHunufTAiBFuZEYpiEuXG30VCDiorSLrdzskKbtSWtgipdZLkme9A%3D%3D&host=r4---sn-4g5ednsz.googlevideo.com&redirect_counter=1&rm=sn-4g5e6d7z&req_id=8f2d5cfda54c36e2&cms_redirect=yes&ipbypass=yes&mip=91.3.45.14&mm=31&mn=sn-h0jeenek&ms=au&mt=1569748179&mv=m&mvi=3&pl=26&lsparams=ipbypass,mip,mm,mn,ms,mv,mvi,pl&lsig=AHylml4wRAIgaLjVzn0DpXNXKSmEhIAGPTx7UVLMXiMzUtsMIfRpHc4CIHbkDZpnvYtZFFAe9mRlHTUuh4JR4kcakW2Ycdlx8jze" > asd
The name is too long, 793 chars total.
Trying to shorten...
New name is videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&.
--2019-09-29 12:13:08--  https://r4---sn-h0jeenek.googlevideo.com/videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&fexp=23842630&c=WEB&txp=3416222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cmime%2Cratebypass%2Cdur%2Clmt&sig=ALgxI2wwRQIhANqYc5fuRTlzb0Lzrivm5_XMLC9arLPxDFM3daHunufTAiBFuZEYpiEuXG30VCDiorSLrdzskKbtSWtgipdZLkme9A%3D%3D&host=r4---sn-4g5ednsz.googlevideo.com&redirect_counter=1&rm=sn-4g5e6d7z&req_id=8f2d5cfda54c36e2&cms_redirect=yes&ipbypass=yes&mip=91.3.45.14&mm=31&mn=sn-h0jeenek&ms=au&mt=1569748179&mv=m&mvi=3&pl=26&lsparams=ipbypass,mip,mm,mn,ms,mv,mvi,pl&lsig=AHylml4wRAIgaLjVzn0DpXNXKSmEhIAGPTx7UVLMXiMzUtsMIfRpHc4CIHbkDZpnvYtZFFAe9mRlHTUuh4JR4kcakW2Ycdlx8jze
Resolving r4---sn-h0jeenek.googlevideo.com (r4---sn-h0jeenek.googlevideo.com)... 172.217.130.41, 2a00:1450:4021::9
Connecting to r4---sn-h0jeenek.googlevideo.com (r4---sn-h0jeenek.googlevideo.com)|172.217.130.41|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 63532218 (61M) [video/mp4]
Saving to: 'videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&'

videoplayback?expire=1569769918&ei=XnWQXY20C   0%[                                                                                              ] 186.26K  --.-KB/s    in 0.1s

2019-09-29 12:13:08 (1.58 MB/s) - Read error at byte 190727/63532218 (Error decoding the received TLS packet.). Retrying.

--2019-09-29 12:13:09--  (try: 2)  https://r4---sn-h0jeenek.googlevideo.com/videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&fexp=23842630&c=WEB&txp=3416222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cmime%2Cratebypass%2Cdur%2Clmt&sig=ALgxI2wwRQIhANqYc5fuRTlzb0Lzrivm5_XMLC9arLPxDFM3daHunufTAiBFuZEYpiEuXG30VCDiorSLrdzskKbtSWtgipdZLkme9A%3D%3D&host=r4---sn-4g5ednsz.googlevideo.com&redirect_counter=1&rm=sn-4g5e6d7z&req_id=8f2d5cfda54c36e2&cms_redirect=yes&ipbypass=yes&mip=91.3.45.14&mm=31&mn=sn-h0jeenek&ms=au&mt=1569748179&mv=m&mvi=3&pl=26&lsparams=ipbypass,mip,mm,mn,ms,mv,mvi,pl&lsig=AHylml4wRAIgaLjVzn0DpXNXKSmEhIAGPTx7UVLMXiMzUtsMIfRpHc4CIHbkDZpnvYtZFFAe9mRlHTUuh4JR4kcakW2Ycdlx8jze
Connecting to r4---sn-h0jeenek.googlevideo.com (r4---sn-h0jeenek.googlevideo.com)|172.217.130.41|:443... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 63532218 (61M), 63341491 (60M) remaining [video/mp4]
Saving to: 'videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&'

videoplayback?expire=1569769918&ei=XnWQXY20C   0%[                                                                                              ] 186.26K  --.-KB/s    in 0s

2019-09-29 12:13:10 (0.00 B/s) - Read error at byte 190727/63532218 (The TLS connection was non-properly terminated.). Retrying.
```

</details>

<details>
<br>

```
wget "https://r4---sn-h0jeenek.googlevideo.com/videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&fexp=23842630&c=WEB&txp=3416222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cmime%2Cratebypass%2Cdur%2Clmt&sig=ALgxI2wwRQIhANqYc5fuRTlzb0Lzrivm5_XMLC9arLPxDFM3daHunufTAiBFuZEYpiEuXG30VCDiorSLrdzskKbtSWtgipdZLkme9A%3D%3D&host=r4---sn-4g5ednsz.googlevideo.com&redirect_counter=1&rm=sn-4g5e6d7z&req_id=8f2d5cfda54c36e2&cms_redirect=yes&ipbypass=yes&mip=91.3.45.14&mm=31&mn=sn-h0jeenek&ms=au&mt=1569748179&mv=m&mvi=3&pl=26&lsparams=ipbypass,mip,mm,mn,ms,mv,mvi,pl&lsig=AHylml4wRAIgaLjVzn0DpXNXKSmEhIAGPTx7UVLMXiMzUtsMIfRpHc4CIHbkDZpnvYtZFFAe9mRlHTUuh4JR4kcakW2Ycdlx8jze"
The name is too long, 817 chars total.
Trying to shorten...
New name is videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&.
--2019-09-29 11:27:54--  https://r4---sn-h0jeenek.googlevideo.com/videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&fexp=23842630&c=WEB&txp=3416222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cmime%2Cratebypass%2Cdur%2Clmt&sig=ALgxI2wwRQIhANqYc5fuRTlzb0Lzrivm5_XMLC9arLPxDFM3daHunufTAiBFuZEYpiEuXG30VCDiorSLrdzskKbtSWtgipdZLkme9A%3D%3D&host=r4---sn-4g5ednsz.googlevideo.com&redirect_counter=1&rm=sn-4g5e6d7z&req_id=8f2d5cfda54c36e2&cms_redirect=yes&ipbypass=yes&mip=91.3.45.14&mm=31&mn=sn-h0jeenek&ms=au&mt=1569748179&mv=m&mvi=3&pl=26&lsparams=ipbypass,mip,mm,mn,ms,mv,mvi,pl&lsig=AHylml4wRAIgaLjVzn0DpXNXKSmEhIAGPTx7UVLMXiMzUtsMIfRpHc4CIHbkDZpnvYtZFFAe9mRlHTUuh4JR4kcakW2Ycdlx8jze
Resolving r4---sn-h0jeenek.googlevideo.com (r4---sn-h0jeenek.googlevideo.com)... 172.217.130.41, 2a00:1450:4021::9
Connecting to r4---sn-h0jeenek.googlevideo.com (r4---sn-h0jeenek.googlevideo.com)|172.217.130.41|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 63532218 (61M) [video/mp4]
Saving to: ‘videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&.1’

videoplayback?expire=1569769918&ei=XnWQXY20C   0%[                                                                                              ] 186.26K  --.-KB/s    in 0.06s

2019-09-29 11:27:54 (3.14 MB/s) - Read error at byte 190727/63532218 (A TLS packet with unexpected length was received.). Retrying.

--2019-09-29 11:27:55--  (try: 2)  https://r4---sn-h0jeenek.googlevideo.com/videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&fexp=23842630&c=WEB&txp=3416222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cmime%2Cratebypass%2Cdur%2Clmt&sig=ALgxI2wwRQIhANqYc5fuRTlzb0Lzrivm5_XMLC9arLPxDFM3daHunufTAiBFuZEYpiEuXG30VCDiorSLrdzskKbtSWtgipdZLkme9A%3D%3D&host=r4---sn-4g5ednsz.googlevideo.com&redirect_counter=1&rm=sn-4g5e6d7z&req_id=8f2d5cfda54c36e2&cms_redirect=yes&ipbypass=yes&mip=91.3.45.14&mm=31&mn=sn-h0jeenek&ms=au&mt=1569748179&mv=m&mvi=3&pl=26&lsparams=ipbypass,mip,mm,mn,ms,mv,mvi,pl&lsig=AHylml4wRAIgaLjVzn0DpXNXKSmEhIAGPTx7UVLMXiMzUtsMIfRpHc4CIHbkDZpnvYtZFFAe9mRlHTUuh4JR4kcakW2Ycdlx8jze
Connecting to r4---sn-h0jeenek.googlevideo.com (r4---sn-h0jeenek.googlevideo.com)|172.217.130.41|:443... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 63532218 (61M), 63341491 (60M) remaining [video/mp4]
Saving to: ‘videoplayback?expire=1569769918&ei=XnWQXY20C87NgQeXzLDICA&ip=138.201.90.189&id=o-ANFEu_crjPlFqVjb94lNz4QrVYX1mGycMvQCUIEcKCQd&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&.1’

videoplayback?expire=1569769918&ei=XnWQXY20C   0%[                                                                                              ] 186.26K  --.-KB/s    in 0s

2019-09-29 11:27:55 (0.00 B/s) - Read error at byte 190727/63532218 (The TLS connection was non-properly terminated.). Retrying.
```

</details>

**Comment by Perflyst** [comment]
Running the wget locally I receive `Connection closed at byte 190727. Retrying.`

**Comment by omarroth** [comment]
Likely duplicate of #750.

**Comment by Perflyst** [comment]
I have a bunch of videos with this error, also it is only on my instance not on invidio.us so not sure if this is the same issue...

**Comment by omarroth** [comment]
I can reproduce the same issue on invidio.us with the same video.

With wget:
<details>
<pre>
<code>
The name is too long, 832 chars total.
Trying to shorten...
New name is videoplayback?expire=1569787443&ei=07mQXeeTGs7n8wSz_peABg&ip=2604:a880:400:d1::b6f:4001&id=o-ABd8vpEM1PGJpaQxW2upAwwqNlMLP1x.
--2019-09-29 10:06:04--  https://r3---sn-pobpb-poql.googlevideo.com/videoplayback?expire=1569787443&ei=07mQXeeTGs7n8wSz_peABg&ip=2604%3Aa880%3A400%3Ad1%3A%3Ab6f%3A4001&id=o-ABd8vpEM1PGJpaQxW2upAwwqNlMLP1xP7u5-QC_okRo4&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&fexp=23842630&c=WEB&txp=3416222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cmime%2Cratebypass%2Cdur%2Clmt&sig=ALgxI2wwRQIhAKEyW9YFnNySdpXEcxH69bQOa5RUsROPWe00EY2JD9lKAiB1BMjNQNWN3_ieJYINv7Qg_9vT7oopyJY4Qc0s61TORQ%3D%3D&host=r4---sn-ab5szn7d.googlevideo.com&redirect_counter=1&rm=sn-ab5e7e7z&req_id=b9ee754a72ca36e2&cms_redirect=yes&ipbypass=yes&mip=137.112.227.229&mm=31&mn=sn-pobpb-poql&ms=au&mt=1569765708&mv=m&mvi=2&pcm2cms=yes&pl=16&lsparams=ipbypass,mip,mm,mn,ms,mv,mvi,pcm2cms,pl&lsig=AHylml4wRAIgRC1xtilY2qSlAJ9Dke3RZhIwas78WflYrsNIkPpm5JcCIFL1UL7iYFaS6aHLJx10SdTpOhhvk00ZiGvw9On9U7IA
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving r3---sn-pobpb-poql.googlevideo.com (r3---sn-pobpb-poql.googlevideo.com)... 149.165.180.206
Connecting to r3---sn-pobpb-poql.googlevideo.com (r3---sn-pobpb-poql.googlevideo.com)|149.165.180.206|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 63532218 (61M) [video/mp4]
Saving to: ‘videoplayback?expire=1569787443&ei=07mQXeeTGs7n8wSz_peABg&ip=2604:a880:400:d1::b6f:4001&id=o-ABd8vpEM1PGJpaQxW2upAwwqNlMLP1x’
<br>
videoplayback?expire=1569787443&ei=07mQXeeTGs7n8wSz_peAB   0%[                                                                                                                                ] 202.26K  --.-KB/s    in 0.1s    
</br>
2019-09-29 10:06:04 (1.37 MB/s) - Read error at byte 207111/63532218 (The TLS connection was non-properly terminated.). Retrying.
</code>
</pre>
</details>

With curl:
<details>
<pre>
<code>
*   Trying 149.165.180.206:443...
* TCP_NODELAY set
* Connected to r3---sn-pobpb-poql.googlevideo.com (149.165.180.206) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.3 (IN), TLS handshake, Server hello (2):
{ [91 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [2767 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [300 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [37 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: C=US; ST=California; L=Mountain View; O=Google LLC; CN=*.googlevideo.com
*  start date: Sep 10 16:26:58 2019 GMT
*  expire date: Nov 19 16:26:58 2019 GMT
*  subjectAltName: host "r3---sn-pobpb-poql.googlevideo.com" matched cert's "*.googlevideo.com"
*  issuer: C=US; O=Google Trust Services; CN=GTS CA 1O1
*  SSL certificate verify ok.
} [5 bytes data]
> GET /videoplayback?expire=1569787443&ei=07mQXeeTGs7n8wSz_peABg&ip=2604%3Aa880%3A400%3Ad1%3A%3Ab6f%3A4001&id=o-ABd8vpEM1PGJpaQxW2upAwwqNlMLP1xP7u5-QC_okRo4&itag=22&source=youtube&requiressl=yes&mime=video%2Fmp4&ratebypass=yes&dur=591.389&lmt=1569466443509032&fvip=4&fexp=23842630&c=WEB&txp=3416222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cmime%2Cratebypass%2Cdur%2Clmt&sig=ALgxI2wwRQIhAKEyW9YFnNySdpXEcxH69bQOa5RUsROPWe00EY2JD9lKAiB1BMjNQNWN3_ieJYINv7Qg_9vT7oopyJY4Qc0s61TORQ%3D%3D&host=r4---sn-ab5szn7d.googlevideo.com&redirect_counter=1&rm=sn-ab5e7e7z&req_id=b9ee754a72ca36e2&cms_redirect=yes&ipbypass=yes&mip=137.112.227.229&mm=31&mn=sn-pobpb-poql&ms=au&mt=1569765708&mv=m&mvi=2&pcm2cms=yes&pl=16&lsparams=ipbypass,mip,mm,mn,ms,mv,mvi,pcm2cms,pl&lsig=AHylml4wRAIgRC1xtilY2qSlAJ9Dke3RZhIwas78WflYrsNIkPpm5JcCIFL1UL7iYFaS6aHLJx10SdTpOhhvk00ZiGvw9On9U7IA HTTP/1.1
> Host: r3---sn-pobpb-poql.googlevideo.com
> User-Agent: curl/7.66.0
> Accept: */*
> 
{ [5 bytes data]
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Last-Modified: Thu, 26 Sep 2019 02:54:03 GMT
< Content-Type: video/mp4
< Date: Sun, 29 Sep 2019 14:09:50 GMT
< Expires: Sun, 29 Sep 2019 14:09:50 GMT
< Cache-Control: private, max-age=20953
< Accept-Ranges: bytes
< Content-Length: 63532218
< Connection: close
< Alt-Svc: quic=":443"; ma=2592000; v="46,43,39"
< X-Content-Type-Options: nosniff
< Server: gvs 1.0
< 
{ [5 bytes data]
* transfer closed with 63325107 bytes remaining to read
* Closing connection 0
} [5 bytes data]
* TLSv1.2 (OUT), TLS alert, close notify (256):
} [2 bytes data]
</code>
</pre>
</details>

**Comment by Perflyst** [comment]
Ok, dunno why I was able to watch the video. Let's track this in #750
